### ISSUE TITLE
Model information setting

### ISSUE THREAD

**Issue by Lee-Gihun** [Lee-Gihun/Pytorch-OSI-setup#1]
model_saver gets model_info dict which should contain

- trained model's class
- dataset info
  - dataset name (and input size)
  - train/valid/test set sizes

- optimization settings
  - optimizer settings
  - scheduler settings

- performance
  - test loss, accuracy

- memo
  - user's custom memo for the model

A method of TrainHandler (info_setter.... maybe?) is desired

https://github.com/Lee-Gihun/Pytorch-OSI-setup/blob/908871bbddb458fb7a75aa2ecdf5c6101a867c8e/train_tools/utils.py#L17-L34

**Comment by kimbob13** [comment]
Need to set experiment name also

**Comment by Lee-Gihun** [comment]
model info dict should contain following keys:
{
model : (model's class name, model's attributes),
dataset_info : (dataset name, validation setting)
optmizier : (optimizer's name, settings)
performance : (test results)
memo : (user's personal memo if necessary)
}

**Comment by Lee-Gihun** [comment]
as like result_logger and plotter, model_saver should get path from user.
(path=self.path)
